src/vs/workbench/services/mappedEdits/browser/mappedEditsService.ts

```diff
@@ -0,0 +1,51 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { CancellationToken } from 'vs/base/common/cancellation';
+import { ITextModel } from 'vs/editor/common/model';
+import { IMappedEditsProvider, IMappedEditsService, MappedEditsContext } from 'vs/workbench/services/mappedEdits/common/mappedEdits';
+import { score } from 'vs/editor/common/languageSelector';
+import { WorkspaceEdit } from 'vs/editor/common/languages';
+import { InstantiationType, registerSingleton } from 'vs/platform/instantiation/common/extensions';
+
+export class MappedEditsService implements IMappedEditsService {
+	readonly _serviceBrand: undefined;
+
+	private readonly _providers = new Set<IMappedEditsProvider>();
```

Comment:
There's the `LanguageFeatureRegistry` that you can use

Reply:
Yes! That allows to score providers based on their selector specificity!